packages/app-config-writer/test/unit/prompt/smartlinks-config.test.ts

```diff
@@ -0,0 +1,300 @@
+import { join } from 'path';
+import { prompt } from 'prompts';
+import type { PromptObject } from 'prompts';
+import * as btp from '@sap-ux/btp-utils';
+import type { ToolsLogger } from '@sap-ux/logger';
+import * as Logger from '@sap-ux/logger';
+import * as store from '@sap-ux/store';
+import { getSmartLinksTargetFromPrompt } from '../../../src';
+import { t } from '../../../src/i18n';
+import * as utils from '../../../src/smartlinks-config/utils';
+import { promptUserPass } from '../../../src/prompt';
+import { yellow } from 'chalk';
+
+jest.mock('prompts', () => ({
+    ...(jest.requireActual('prompts') as object),
+    prompt: jest.fn()
+}));
+const promptMock = prompt as jest.Mock;
+
+jest.mock('@sap-ux/btp-utils', () => ({
+    ...jest.requireActual('@sap-ux/btp-utils'),
+    isAppStudio: jest.fn(),
+    listDestinations: jest.fn()
+}));
+
+const loggerMock: ToolsLogger = {
+    debug: jest.fn(),
+    info: jest.fn(),
+    warn: jest.fn(),
+    error: jest.fn()
+} as Partial<ToolsLogger> as ToolsLogger;
+jest.spyOn(Logger, 'ToolsLogger').mockImplementation(() => loggerMock);
+
+const serviceMock = { read: jest.fn() };
+let isAppStudioMock: jest.SpyInstance;
+let listDestinationsMock: jest.SpyInstance;
+
+describe('Test function getSmartLinksTargetFromPrompt', () => {
+    // Mock setup
+    const debugMock = loggerMock.debug as unknown as jest.SpyInstance;
+    let getSystemCredentialsSpy: jest.SpyInstance;
+    let getServiceMock: jest.SpyInstance;
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        isAppStudioMock = jest.spyOn(btp, 'isAppStudio');
+        getSystemCredentialsSpy = jest.spyOn(utils, 'getLocalStoredCredentials');
+        getServiceMock = jest.spyOn(store, 'getService').mockImplementation(() => serviceMock as any);
+        listDestinationsMock = jest.spyOn(btp, 'listDestinations');
+    });
+
+    describe('Prompt for target url and client: ', () => {
```

Comment:
I am not sure `.matchSnapshot` is a good choice here. Please think about checking against the specific object properties that are relevant in each test. It makes it easier for a reader of the test to understand what you are verifying.

Reply:
Reworked to check for the expected properties directly.